tests/tasks/test_testing.py

```diff
@@ -71,3 +79,50 @@ def test_dummy_timestamp_feature_task(dummy_eopatch: EOPatch):
 
     eopatch3 = task.execute(dummy_eopatch.copy(), seed=11)
     assert eopatch1 != eopatch3
+
+
+@pytest.mark.parametrize(
+    ("feature_type", "shape", "dtype", "distribution"),
+    [
+        (FeatureType.DATA, (10, 20, 21, 5), np.float32, UniformDistribution(-3, -1.5)),
+        (FeatureType.DATA_TIMELESS, (83, 69, 1), int, UniformDistribution(-2, 5)),
+        (FeatureType.MASK, (10, 4, 16, 7), np.int8, UniformDistribution(-3, 7)),
+        (FeatureType.LABEL, (10, 271), bool, UniformDistribution(0, 1)),
+        (FeatureType.DATA, (7, 20, 31, 1), np.float32, NormalDistribution(-3, 1.5)),
+        (FeatureType.DATA_TIMELESS, (30, 61, 2), float, NormalDistribution(-3422.23, 1522)),
+        (FeatureType.DATA_TIMELESS, (83, 69, 10), int, NormalDistribution(-2, 10)),
+    ],
+)
+@pytest.mark.parametrize("seed", [1, 22, 9182])
+def test_generate_raster_feature_task(dummy_eopatch, feature_type, shape, dtype, distribution, seed):
+    dummy_eopatch.timestamps = ["2011-08-12"] * 10
+    feature = feature_type, "FEATURE"
+
+    task = GenerateRasterFeatureTask(feature, shape=shape, dtype=dtype, distribution=distribution)
+    eopatch = task.execute(dummy_eopatch.copy(), seed=seed)
+
+    assert feature in eopatch
+    assert len(eopatch.get_features()) == 3
+
+    data: np.ndarray = eopatch[feature]
+    assert data.shape == shape
+    assert data.dtype == dtype
+
+    if is_discrete_type(data.dtype):
+        if isinstance(distribution, UniformDistribution):
+            _, p_value = chisquare(np.unique(data, return_counts=True)[1])
+            assert p_value > 0.05
+        else:
+            # "normally distributed" integers are just a hack, so we test what we can
+            assert np.mean(data) == pytest.approx(distribution.mean, abs=0.2)
+            assert np.std(data) == pytest.approx(distribution.std, abs=0.2)
```

Comment:
should the error used here be relative and not absolute? Just in case the tests change or something

Reply:
Additionally, this could in principle also be tested with the chi-square, but due to the complicated nature of the rounded numbers from the normal distribution let's not go there..

Reply:
yeah, i'd rather not calculate the correct distribution for this by hand :see_no_evil:

Reply:
for the relative vs. absolute, i'd argue absolute, because there is no reason the `mean` should be more accurate if it's closer to 0

Reply:
I guess we can keep both at `abs`, but we replace the hard-coded values with the expected standard error of the mean and of the st. dev, based on the number of samples.

If I'm not mistaken, the equations are:

```math
SE(\bar{x}) = \frac{\sigma_{x}}{\sqrt{n}}
```

```math
SE(\sigma_{x}) = \frac{\sigma_{x}}{\sqrt{2(n-1)}}
```

These are 95% confidence intervals and can be used for the `abs` value in the approximations.

Taken from https://www.tqmp.org/RegularArticles/vol10-2/p107/p107.pdf

can you compare these values to the 0.2 based on the data from the test?